src/CMakeLists.txt

```diff
@@ -10,12 +10,21 @@ set(ENABLE_SINC_FAST_CONVERTER ${LIBSAMPLERATE_ENABLE_SINC_FAST_CONVERTER} PAREN
 set(ENABLE_SINC_MEDIUM_CONVERTER ${LIBSAMPLERATE_ENABLE_SINC_MEDIUM_CONVERTER} PARENT_SCOPE)
 set(ENABLE_SINC_BEST_CONVERTER ${LIBSAMPLERATE_ENABLE_SINC_BEST_CONVERTER} PARENT_SCOPE)
 
+include(CMakePushCheckState)
+include(CheckCSourceCompiles)
 include(ClipMode)
 include(CMakePackageConfigHelpers)
 
 # This will set CPU_CLIPS_NEGATIVE and CPU_CLIPS_POSITIVE
 clip_mode()
 
+# check for symbol visibility attributes support
+cmake_push_check_state()
+set(CMAKE_REQUIRED_FLAGS "-fvisibility=hidden -Werror")
```

Comment:
...does this actually work on MSVC?

Reply:
> ...does this actually work on MSVC?

Of course not. I just avoided multiple if()/endif()s.

Reply:
how do we get this to work for `__declspec(dllexport)` at a later stage then? add another `#elif defined (_WIN32) || defined (__CYGWIN__)` macro clause?

Reply:
> how do we get this to work for `__declspec(dllexport)` at a later stage then? add another `#elif defined (_WIN32) || defined (__CYGWIN__)` macro clause?

We don't. src/CMakeLists.txt uses the def file for exports:
https://github.com/libsndfile/libsamplerate/blob/master/src/CMakeLists.txt#L19

Reply:
oh right, sorry, forgot that already :rofl: